build.js

```diff
@@ -1,670 +1,191 @@
-      download();
-    }
-  }
-
-  function updatePost(error) {
-    if (postDone) {
+        console.log("Development size: %d KB; compressed size: %d KB",
+          fileSizes.src, fileSizes.dest);
+      });
+    }]
+  }, function afterBuild(err) {
+    if (err) {
+      process.exit(1);
       return;
     }
-    if (error) {
-      postDone = true;
-      return finalize(error);
-    }
-    postPending--;
-    if (!postPending) {
-      postDone = true;
-      finalize();
-    }
-  }
-}
-
-// Generates the navigation section for a Markdown document.
-// Ported from `mdtoc.rb` by Sam Stephenson.
-function genNav(source) {
-  var headers = [];
-  var lines = source.split(/\r?\n/);
-  // First pass: Scan the Markdown source looking for titles of the format:
-  // `### Title ###`. Record the line number, header level (number of
-  // octothorpes), and text of each matching title.
-  lines.forEach(function (line, index) {
-    var match = /^(\#{1,6})\s+(.+?)\s+\1$/.exec(line);
-    if (match) {
-      headers.push([index, match[1].length, match[2]]);
-    }
-  });
-  // Second pass: Iterate over all matched titles and compute their
-  // corresponding section numbers. Then replace the titles with annotated
-  // anchors.
-  var lastSection, lastLevel;
-  headers.forEach(function (value) {
-    var index = value[0], level = value[1], text = value[2], section, length;
-    if (lastSection) {
-      // Clone the last section metadata array.
-      section = lastSection.slice(0);
-      if (lastLevel < level) {
-        section.push(1);
-      } else {
-        length = lastLevel - level;
-        while (length--) {
-          section.pop();
-        }
-        section[section.length - 1] += 1;
-      }
-    } else {
-      section = [1];
-    }
-    lines[index] = Array(level + 1).join("#") + "<a name=\"section_" + section.join(".") + "\"></a>" + text;
-    value.push(section);
-    lastSection = section;
-    lastLevel = level;
-  });
-  // Third pass: Iterate over matched titles once more to produce the table of
-  // contents.
-  var navigation = headers.map(function (value) {
-    var index = value[0], level = value[1], text = value[2], section = value[3], name = section.join(".");
-    return "<li><a href=\"#section_" + name + "\">" + text + "</a></li>";
+    process.exit(0);
   });
-  navigation.push("");
-  return {
-    "navigation": navigation,
-    "lines": lines
-  };
 }
 
-function buildPage(title, template, src, options) {
-  var parts = genNav(src);
-  return template.replace(/<%=\s*(.+?)\s*%>/g, function interpolate(match, section) {
-    if (section == "title") {
-      return title;
-    }
-    if (section == "navigation") {
-      // Insert the table of contents directly into the template.
-      return parts.navigation.join("\n");
-    }
-    if (section == "source") {
-      // Convert the page to HTML and insert it into the body.
-      return marked(parts.lines.join("\n"), options);
+function writePages(pages, callback) {
```

Comment:
The inner callback to `eachPage` will be called three times, but the outer one wasn't called at all. I renamed the inner callbacks to `next*` for clarity.